modules/compatibility/src/test/java/org/apache/ignite/compatibility/sql/model/City.java

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright 2020 GridGain Systems, Inc. and Contributors.
+ *
+ * Licensed under the GridGain Community Edition License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.gridgain.com/products/software/community-edition/gridgain-community-edition-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.compatibility.sql.model;
+
+import java.util.Arrays;
+import java.util.Random;
+import org.apache.ignite.cache.QueryEntity;
+import org.apache.ignite.cache.QueryIndex;
+import org.apache.ignite.cache.QueryIndexType;
+import org.apache.ignite.cache.query.annotations.QuerySqlField;
+
+import static java.util.Collections.singletonList;
+import static org.apache.ignite.compatibility.sql.model.Country.Factory.COUNTRY_CNT;
+import static org.apache.ignite.compatibility.sql.model.ModelUtil.randomString;
+
+/**
+ * City model.
+ */
+public class City {
+    /** */
+    @QuerySqlField
+    private final String name;
+
+    /** */
+    @QuerySqlField
+    private final String zipCode;
+
+    /** */
+    @QuerySqlField
+    private final int countryId;
+
+    /** */
+    @QuerySqlField
+    private final int population;
+
+    /** */
+    public City(String name, String zipCode, int countryId, int population) {
+        this.name = name;
+        this.zipCode = zipCode;
+        this.countryId = countryId;
+        this.population = population;
+    }
+
+    /** */
+    public String name() {
+        return name;
+    }
+
+    /** */
+    public String zipCode() {
+        return zipCode;
+    }
+
+    /** */
+    public int countryId() {
+        return countryId;
+    }
+
+    /** */
+    public int population() {
+        return population;
+    }
+
+    /** */
+    public static class Factory implements ModelFactory {
```

Comment:
Factory classes looks identically, let's extract base\abstract class.

BaseFactory<T>
```suggestion
    public static class BaseFactory implements ModelFactory {
   
   protected Random rnd;
   protected final qryEntity;
   
    public BaseFactory(QueryEntity ent, String tableName, int count) {
    ...
    }
   
    public abstract T createRandom();
   
    public void init(int seed) {
            rnd = new Random(seed);
     }
   
     public QueryEntity queryEntity() {
            return qryEntity;
     }

   } 
```

Reply:
👌